### PR TITLE
Added filtersets and search filters for all models

### PR DESCRIPTION
**Pull request by IegorSoboliev**

### CODE REVIEW

File: business_register/filters.py [hunk 1]
```diff
@@ -14,6 +16,14 @@ class Meta:
         }
 
 
+class FopFilterSet(filters.FilterSet):
+    fullname = filters.CharFilter(lookup_expr='iexact')
```
Comment (roman-tiukh) [inline business_register/filters.py:20]: why not icontains?

File: business_register/filters.py [hunk 2]
```diff
@@ -14,6 +16,14 @@ class Meta:
...
+    address = filters.CharFilter(lookup_expr='icontains')
+
+    class Meta:
+        model = Fop
+
```
Comment (roman-tiukh) [inline business_register/filters.py:25]: you need to add blank list of fields
```suggestion
    class Meta:
        model = Fop
        fields = ()
```

**Comment by IegorSoboliev** [comment]
Please, review this and then I will continue with other models

File: business_register/views/rfop_views.py [hunk 3]
```diff
@@ -27,4 +30,8 @@ def retrieve(self, request, pk=None):
 class FopView(CachedViewMixin, viewsets.ReadOnlyModelViewSet):
     queryset = Fop.objects.all()
     serializer_class = FopSerializer
+    filter_backends = (DjangoFilterBackend, SearchFilter)
+    filterset_class = FopFilterSet
+    search_fields = ('fullname', 'address', 'registration_date', 'authority')
```
Comment (roman-tiukh) [inline business_register/views/rfop_views.py:35]: authority not a char field

File: business_register/filters.py (hunk 1 above)
Comment (roman-tiukh) [reply business_register/filters.py:20]: i think icontains be better here

Comment (IegorSoboliev) [reply business_register/filters.py:20]: there are a lot of people with the same full name, not mentioning with similar)

File: business_register/filters.py (hunk 2 above)
Comment (IegorSoboliev) [reply business_register/filters.py]: done

File: business_register/views/rfop_views.py (hunk 3 above)
Comment (IegorSoboliev) [reply business_register/views/rfop_views.py]: fixed

File: business_register/views/company_views.py [hunk 4]
```diff
@@ -9,9 +11,13 @@
 
 class CompanyView(CachedViewMixin, viewsets.ReadOnlyModelViewSet):
     permission_classes = [AllowAny]
-    queryset = Company.objects.all()
+    queryset = Company.objects.select_related('parent', 'company_type', 'status', 'authority',
+                                              'bylaw') \
+        .prefetch_related('founders', 'predecessors', 'assignees', 'signers', 'kveds',
+                          'bancruptcy_readjustment', 'company_detail', 'exchange_data',
+                          'termination_started').all()
```
Comment (roman-tiukh) [inline business_register/views/company_views.py:18]: so much prettier
```suggestion
    queryset = Company.objects.select_related(
        'parent', 'company_type', 'status', 'authority',  'bylaw'
    ).prefetch_related(
        'founders', 'predecessors', 'assignees', 'signers', 'kveds',
        'bancruptcy_readjustment', 'company_detail', 'exchange_data', 'termination_started'
    ).all()
```

File: business_register/views/rfop_views.py [hunk 5]
```diff
@@ -25,6 +27,10 @@ def retrieve(self, request, pk=None):
 
 
 class FopView(CachedViewMixin, viewsets.ReadOnlyModelViewSet):
-    queryset = Fop.objects.all()
+    queryset = Fop.objects.select_related('status', 'authority').prefetch_related('kveds',
+                                                                                  'exchange_data'). \
+        all()
```
Comment (roman-tiukh) [inline business_register/views/rfop_views.py:32]: doesn't look pretty
```suggestion
    queryset = Fop.objects.select_related(
        'status', 'authority'
    ).prefetch_related(
        'kveds', 'exchange_data'
    ).all()
```

File: data_ocean/views.py [hunk 6]
```diff
@@ -44,9 +44,9 @@ def get(self, request):
 class RegisterView(viewsets.ReadOnlyModelViewSet):
     queryset = Register.objects.all()
     serializer_class = RegisterSerializer
-    filterset_class = RegisterFilter
+    filterset_class = RegisterFilterSet
     filter_backends = [DjangoFilterBackend, SearchFilter]
-    search_fields = ['name', 'source_register_id']
+    search_fields = ['name', 'source_register_id', 'source_name']
```
Comment (roman-tiukh) [inline data_ocean/views.py:49]: I think you only need to leave only "name".
its better as search.
```suggestion
    search_fields = ['name']
```

File: location_register/views/ratu_viewsets.py [hunk 7]
```diff
@@ -1,29 +1,39 @@
...
-    
-class StreetView(CachedViewMixin, viewsets.ReadOnlyModelViewSet):
-    queryset = RatuStreet.objects.all()
-    serializer_class = StreetSerializer
+    serializer_class = RatuCityDistrictSerializer
+
 
+class RatuStreetView(CachedViewMixin, viewsets.ReadOnlyModelViewSet):
+    queryset = RatuStreet.objects.select_related('region', 'district', 'city', 'citydistrict') \
+        .all()
```
Comment (roman-tiukh) [inline location_register/views/ratu_viewsets.py:35]:
```suggestion
    queryset = RatuStreet.objects.select_related(
        'region', 'district', 'city', 'citydistrict'
    ).all()
```

File: business_register/views/rfop_views.py (hunk 5 above)
Comment (IegorSoboliev) [reply business_register/views/rfop_views.py:32]: done

File: location_register/views/drv_viewsets.py [hunk 8]
```diff
@@ -1,11 +1,17 @@
...
+from location_register.filters import DrvBuildingFilterSet
 from location_register.models.drv_models import DrvBuilding
 from location_register.serializers.drv_serializers import DrvBuildingSerializer
-from rest_framework import generics, viewsets
 
 
 class DrvBuildingViewSet(CachedViewMixin, viewsets.ReadOnlyModelViewSet):
-    queryset = DrvBuilding.objects.all()
+    queryset = DrvBuilding.objects.select_related('region', 'district', 'council', 'ato', 'street',
+                                                  'zip_code', ).all()
```
Comment (roman-tiukh) [inline location_register/views/drv_viewsets.py:13]: more pretty )
```suggestion
    queryset = DrvBuilding.objects.select_related(
        'region', 'district', 'council', 'ato', 'street', 'zip_code',
    ).all()
```

File: business_register/views/company_views.py [hunk 9]
```diff
@@ -9,9 +11,14 @@
...
-    queryset = Company.objects.all()
+    queryset = Company.objects.select_related(
+        'parent', 'company_type', 'status', 'authority',  'bylaw'
+    ).prefetch_related(
+        'founders', 'predecessors', 'assignees', 'signers', 'kveds',
+        'bancruptcy_readjustment', 'company_detail', 'exchange_data', 'termination_started'
+    ).all()
     serializer_class = CompanySerializer
-    filter_backends = (DjangoFilterBackend,)
+    filter_backends = (DjangoFilterBackend, SearchFilter)
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply business_register/views/company_views.py:21]: added

File: location_register/views/ratu_viewsets.py [hunk 10]
```diff
@@ -1,29 +1,40 @@
...
+
 
+class RatuStreetView(CachedViewMixin, viewsets.ReadOnlyModelViewSet):
+    queryset = RatuStreet.objects.select_related(
+        'region', 'district', 'city', 'citydistrict'
+    ).all()
+    serializer_class = RatuStreetSerializer
+    filter_backends = (DjangoFilterBackend, SearchFilter)
+    filterset = RatuStreetFilterSet
+    search_fields = 'name'
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply location_register/views/ratu_viewsets.py:36]: fixed

File: location_register/views/drv_viewsets.py (hunk 8 above)
Comment (IegorSoboliev) [reply location_register/views/drv_viewsets.py]: done

File: business_register/views/company_views.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply business_register/views/company_views.py:19]: fixed

File: location_register/views/ratu_viewsets.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply location_register/views/ratu_viewsets.py]: done

File: location_register/views/drv_viewsets.py (hunk 8 above)
Comment (IegorSoboliev) [reply location_register/views/drv_viewsets.py]: done

File: business_register/views/company_views.py [hunk 11]
```diff
@@ -5,14 +5,20 @@
...
 
 
 class CompanyView(CachedViewMixin, viewsets.ReadOnlyModelViewSet):
     permission_classes = [AllowAny]
-    queryset = Company.objects.all()
+    queryset = Company.objects.select_related('parent', 'company_type', 'status', 'authority',
+                                              'bylaw') \
+        .prefetch_related('founders', 'predecessors', 'assignees', 'signers', 'kveds',
+                          'bancruptcy_readjustment', 'company_detail', 'exchange_data',
+                          'termination_started').all()
```
Comment (roman-tiukh) [inline business_register/views/company_views.py:17]: One more little change))
```suggestion
    queryset = Company.objects.select_related(
        'parent', 'company_type', 'status', 'authority', 'bylaw',
    ).prefetch_related(
        'founders', 'predecessors', 'assignees', 'signers', 'kveds', 'termination_started',
        'bancruptcy_readjustment', 'company_detail', 'exchange_data',
    ).all()
```

**Pull request merged**